.github/workflows/ci.yml

```diff
@@ -122,3 +122,33 @@ jobs:
       working-directory: ${{github.workspace}}/build
       shell: bash -l {0}
       run: ctest --output-on-failure -C $BUILD_TYPE
+
+  # Job preoducing code coverage report
+  # =========================================
+  Code_coverage:
+    runs-on: macOS-10.15
+
+    steps:
+    - name: "Install libraries"
+      run: brew install boost hdf5 eigen lcov
+
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
+
+    - name: Build and test for code coverage
+      run: |
+        cmake -E make_directory $GITHUB_WORKSPACE/build
+        cd $GITHUB_WORKSPACE/build
+        cmake $GITHUB_WORKSPACE \
+          -DCMAKE_BUILD_TYPE=$BUILD_TYPE \
+          -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR \
+          -DHIGHFIVE_USE_EIGEN:BOOL=TRUE \
+          -DHIGHFIVE_BUILD_DOCS:BOOL=FALSE \
+          -DCMAKE_CXX_FLAGS="-coverage -O0"
```

Comment:
Instead of repeating all options, use the `CMAKE_OPTIONS` variable (see above)

Reply:
the whole thing with quoted arguments is just a mess, I hope the way I solved is ok.